File: c/src/Creational/Factory_Method/TestSoupFactoryMethod.c
0


//TestSoupFactoryMethod.java - testing the factory method

class TestSoupFactoryMethod {         

   public static void main(String[] args) {
       
       SoupFactoryMethod soupFactoryMethod = 
         new SoupFactoryMethod();
       SoupBuffet soupBuffet = 
         soupFactoryMethod.makeSoupBuffet();
       soupBuffet.setSoupBuffetName(
         soupFactoryMethod.makeBuffetName());
       soupBuffet.setChickenSoup(
         soupFactoryMethod.makeChickenSoup());
       soupBuffet.setClamChowder(
         soupFactoryMethod.makeClamChowder()); 
       soupBuffet.setFishChowder(
         soupFactoryMethod.makeFishChowder());
       soupBuffet.setMinnestrone(
         soupFactoryMethod.makeMinnestrone());
       soupBuffet.setPastaFazul(
         soupFactoryMethod.makePastaFazul()); 
       soupBuffet.setTofuSoup(
         soupFactoryMethod.makeTofuSoup());
       soupBuffet.setVegetableSoup(
          soupFactoryMethod.makeVegetableSoup()); 
       System.out.println("At the  " + 
                           soupBuffet.getSoupBuffetName() + 
                           soupBuffet.getTodaysSoups());
       
       
       SoupFactoryMethod bostonSoupFactoryMethod = 
         new BostonSoupFactoryMethodSubclass();
       SoupBuffet bostonSoupBuffet = 
         bostonSoupFactoryMethod.makeSoupBuffet();
       bostonSoupBuffet.setSoupBuffetName(
         bostonSoupFactoryMethod.makeBuffetName());
       bostonSoupBuffet.setChickenSoup(
         bostonSoupFactoryMethod.makeChickenSoup());
       bostonSoupBuffet.setClamChowder(
         bostonSoupFactoryMethod.makeClamChowder());
       bostonSoupBuffet.setFishChowder(
         bostonSoupFactoryMethod.makeFishChowder());
       bostonSoupBuffet.setMinnestrone(
         bostonSoupFactoryMethod.makeMinnestrone());
       bostonSoupBuffet.setPastaFazul(
         bostonSoupFactoryMethod.makePastaFazul());
       bostonSoupBuffet.setTofuSoup(
         bostonSoupFactoryMethod.makeTofuSoup());
       bostonSoupBuffet.setVegetableSoup(
         bostonSoupFactoryMethod.makeVegetableSoup()); 
       System.out.println("At the  " + 
                           bostonSoupBuffet.getSoupBuffetName() + 
                           bostonSoupBuffet.getTodaysSoups());
       
       SoupFactoryMethod honoluluSoupFactoryMethod = 
         new HonoluluSoupFactoryMethodSubclass();
       SoupBuffet honoluluSoupBuffet = 
         honoluluSoupFactoryMethod.makeSoupBuffet();
       honoluluSoupBuffet.setSoupBuffetName(
         honoluluSoupFactoryMethod.makeBuffetName());
       honoluluSoupBuffet.setChickenSoup(
         honoluluSoupFactoryMethod.makeChickenSoup());
       honoluluSoupBuffet.setClamChowder(
         honoluluSoupFactoryMethod.makeClamChowder()); 
       honoluluSoupBuffet.setFishChowder(
         honoluluSoupFactoryMethod.makeFishChowder());
       honoluluSoupBuffet.setMinnestrone(
         honoluluSoupFactoryMethod.makeMinnestrone());
       honoluluSoupBuffet.setPastaFazul(
         honoluluSoupFactoryMethod.makePastaFazul()); 
       honoluluSoupBuffet.setTofuSoup(
         honoluluSoupFactoryMethod.makeTofuSoup());
       honoluluSoupBuffet.setVegetableSoup(
         honoluluSoupFactoryMethod.makeVegetableSoup()); 
       System.out.println("At the  " + 
                           honoluluSoupBuffet.getSoupBuffetName() + 
                           honoluluSoupBuffet.getTodaysSoups());
     }
}      

File: java/t/test/Creational/TestSoupFactoryMethod.java
//TestSoupFactoryMethod.java - testing the factory method

package test.Creational;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Creational.Factory_Method.*;
import common.*;



class TestSoupFactoryMethod {         

	public void thetester( SoupFactoryMethod sfm, String msg)
	{
       SoupBuffet soupBuffet = sfm.makeSoupBuffet();
       soupBuffet.setSoupBuffetName( sfm.makeBuffetName());
       soupBuffet.setChickenSoup( sfm.makeChickenSoup());
       soupBuffet.setClamChowder( sfm.makeClamChowder()); 
       soupBuffet.setFishChowder( sfm.makeFishChowder());
       soupBuffet.setMinnestrone( sfm.makeMinnestrone());
       soupBuffet.setPastaFazul( sfm.makePastaFazul()); 
       soupBuffet.setTofuSoup( sfm.makeTofuSoup());
       soupBuffet.setVegetableSoup( sfm.makeVegetableSoup()); 
       org.junit.Assert.assertEquals(msg,
			"",
			 ("At the  " + soupBuffet.getSoupBuffetName() + soupBuffet.getTodaysSoups())
		);
	}
       

	@Test
   public void base_sfm() {
		thetester( new SoupFactoryMethod(), "Base Soup Factory Method");
	}
       
	@Test
   public void boston_sfm() {
		thetester( new BostonSoupFactoryMethodSubclass(), "Boston Soup Factory Method");
	}
       
	@Test
   public void hono_sfm() {
		thetester( new HonoluluSoupFactoryMethodSubclass(), "Honolulu Soup Factory Method");
	}
}      
